backend/matching/src/services/get-match-items.ts

```diff
@@ -1,7 +1,133 @@
-export const getMatchItems = () => {
-  // TODO: Get Match Items
-  return {
-    roomId: '',
-    questionId: '',
-  };
+import { IMatchType } from "@/types";
+import axios from 'axios';
+
+interface IGetRandomQuestionPayload {
+  attemptedQuestions: number[];
+  difficulty?: string;
+  topic?: string;
+}
+
+
+interface IServiceResponse<T> {
+    success: boolean;
+    data?: T;
+    error?: { message: string };
+  }
+
+interface IQuestion {
+  id: number;
+  title: string;
+  description: string;
+  difficulty: string;
+  topic: string[];
+}
+
+interface IMatchItemsResponse {
+    roomName: string;
+    questionId: number;
+    question: IQuestion;
+}
+
+export const getMatchItems = async (
+    searchIdentifier: IMatchType,
+    topic?: string,
+    difficulty?: string,
+    userId1?: string,
+    userId2?: string
+): Promise<IMatchItemsResponse | undefined> => {
+  const userEndpoint = `${process.env.USER_SERVER_ENDPOINT}`;
+  const questionEndpoint = `${process.env.QUESTION_SERVER_ENDPOINT}`;
+  const collabServerEndpoint = `${process.env.COLLAB_SERVER_ENDPOINT}`;
+
+  try {
+    if (!userId1 || !userId2) {
+      throw new Error('Both user IDs are required');
+    }
+
+    // Fetch attempted questions for both users
+    const [attemptedQuestions1, attemptedQuestions2] = await Promise.all([
+      fetchAttemptedQuestions(userEndpoint, userId1),
+      fetchAttemptedQuestions(userEndpoint, userId2)
+    ]);
+
+    // Combine attempted questions from both users
+    const allAttemptedQuestions = [...new Set([...attemptedQuestions1, ...attemptedQuestions2])];
+
+    // Prepare payload for the /random endpoint
+    const payload: IGetRandomQuestionPayload = {
+      attemptedQuestions: allAttemptedQuestions,
+    };
+
+    if (searchIdentifier === 'difficulty' && difficulty) {
+      payload.difficulty = difficulty;
+    } else if (searchIdentifier === 'topic' && topic) {
+      payload.topic = topic;
+    } else if (searchIdentifier === 'exact match' && topic && difficulty) {
+      payload.topic = topic;
+      payload.difficulty = difficulty;
+    }
+
+    // Query the question endpoint using the /random endpoint
+    const questionResponse = await axios.post<IServiceResponse<{ question: IQuestion }>>(
+      `${questionEndpoint}/questions/random`,
```

Comment:
define all the routes and service clients being called in a separate services file for better maintainability